### PR TITLE
rgw_file: fix assert upon setattr on bucket

### PR DESCRIPTION
**Pull request by guihecheng**
Fixes: http://tracker.ceph.com/issues/20287

Signed-off-by: Gui Hecheng <guihecheng@cmss.chinamobile.com>

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@ceph-jenkins this happens when (on master) we're doing setattr on a bucket with nothing in it, apparently (verified)

**Comment by alfredodeza** [comment]
OK to test

**Comment by mattbenjamin** [comment]
got delayed a bit, can't build downstream without LDFLAGS="-lasan" (whaaa?)

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -6259,9 +6259,8 @@ void RGWSetAttrs::execute()
 
   rgw_obj obj(s->bucket, s->object);
 
-  store->set_atomic(s->obj_ctx, obj);
-
   if (!s->object.empty()) {
+    store->set_atomic(s->obj_ctx, obj);
```
Comment (mattbenjamin) [inline src/rgw/rgw_op.cc]: ah, ok, this is correct, the else case calls rgw_bucket_set_attrs

**Pull request merged**